### PR TITLE
Update to using spack-stack installed libraries (except wcoss2)

### LINKED ISSUE #21: Add Hercules support
The new MSU partition Hercules has come online. Add support for Hercules.

See https://github.com/NOAA-EMC/global-workflow/issues/1588 for more details.

### LINKED ISSUE #22: Upgrade non-production libraries to spack-stack
The library versions/compiler should be upgraded to spack-stack/Intel 2021+ on each research platform to match the global workflow and move away from the older Intel 18 compilers.

### PR DESCRIPTION
**Pull request by aerorahul**
This PR:
1. Is into **dev/gfsv17** and **not** **develop** or any of the **release** branches.
2. adds `hercules` to the platform list
3. builds using **spack-stack v1.6.0** installed libraries
4. introduces `build.platform.ver` for the various platforms and respective `run.platform.ver`.  `run.platform.ver` is currently not exercised.
5. updates library versions for all platforms **except** `wcoss2`

The build was tested on:
- [x] WCOSS2
- [x] Hera
- [x] Orion
- [x] Hercules
- [x] Jet

For those who wish to try to build:
```bash
INSTALL_TARGET=platform ./build.sh
```
where `platform = hera|hercules|orion|jet|wcoss2`

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -3,37 +3,44 @@
...
 MODULEFILE_INSTALL_PREFIX=${MODULEFILE_INSTALL_PREFIX:-"${INSTALL_PREFIX}/modulefiles"}
 
-target=$(echo $INSTALL_TARGET | tr [:upper:] [:lower:])
-if [[ "$target" =~ ^(wcoss2|hera|orion|jet)$ ]]; then
-  source $pkg_root/versions/build.ver
+target="${INSTALL_TARGET,,}"
+if [[ "${target}" =~ ^(wcoss2|hera|orion|jet|hercules)$ ]]; then
+  # copy the target specific build.ver and run.ver
+  cp "${pkg_root}/versions/build.${target}.ver" "${pkg_root}/versions/build.ver"
+  cp "${pkg_root}/versions/run.${target}.ver" "${pkg_root}/versions/run.ver"
```
Comment (DavidHuber-NOAA) [inline build.sh]: Links would probably be better here.

File: modulefiles/prepobs_hera.lua [hunk 2]
```diff
@@ -2,12 +2,15 @@ help([[
 Load environment to build prepobs on Hera
 ]])
 
-load("cmake/3.20.1")
+prepend_path("MODULEPATH", "/scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.5.1/envs/gsi-addon/install/modulefiles/Core")
```
Comment (DavidHuber-NOAA) [inline modulefiles/prepobs_hera.lua]: Should this be bumped to 1.6.0 since that's where the global workflow is headed?

**Review by DavidHuber-NOAA — CHANGES_REQUESTED** [review]
Looks pretty good, though I wonder if we should be upgrading to SS/1.6.0.  I could be convinced either way.

**Comment by KateFriedman-NOAA** [comment]
Build tested successfully on WCOSS2. See this build log on Dogwood: `/lfs/h2/emc/global/noscrub/kate.friedman/git/OBSPROC/feature-spack-stack/build.log`

File: build.sh (hunk 1 above)
Comment (aerorahul) [reply build.sh]: done.

File: modulefiles/prepobs_hera.lua (hunk 2 above)
Comment (aerorahul) [reply modulefiles/prepobs_hera.lua]: yep. done.

**Comment by DavidHuber-NOAA** [comment]
@aerorahul The MKL libraries need to be loaded separately on Hercules.  This is done with `module load intel-oneapi-mkl/2023.1.0`.

File: build.sh (hunk 1 above)
Comment (KateFriedman-NOAA) [reply build.sh]: Doesn't ops require copies and not symlinks?

**Comment by DavidHuber-NOAA** [comment]
Test build was successful on Jet.

File: build.sh (hunk 1 above)
Comment (aerorahul) [reply build.sh]: copying instead of linking.  can update later if necessary. Ok?

Comment (KateFriedman-NOAA) [reply build.sh]: Sure

**Review by KateFriedman-NOAA — APPROVED** [review]
Looks good, thanks @aerorahul !

**Pull request merged**